lib/netconf.js

```diff
@@ -185,26 +185,23 @@ class Client extends EventEmitter {
     }
 
     /**
-     * Closes the SSH connection to netconf, emits close event
+     * Closes the SSH connection to netconf
      * @function close
      * @param  {Function} callback Callback function fired after netconf session ends
      */
-    close(callback){
+    close(callback) {
         const self = this;
 
         // send close-session request to netconf
         self.rpc('close-session', (err, reply) => {
-            // close out SSH tunnel
-            self.sshConn.end();
-            self.connected = false;
-            self.netconf = void 0;
-
-            // if callback funciton passed, continue with callback
-            if(typeof callback === 'function'){
-                callback();
+            if (!callback) {
+                return;
             }
 
-            self.emit('close');
+            if (err) {
```

Comment:
Add type check for callback, it is more accurate than just the truthy check, and can prevent errors